internal/grpc/services/userprovider/userprovider.go

```diff
@@ -59,12 +70,40 @@ func parseConfig(m map[string]interface{}) (*config, error) {
 	return c, nil
 }
 
-func getDriver(c *config) (user.Manager, error) {
-	if f, ok := registry.NewFuncs[c.Driver]; ok {
-		return f(c.Drivers[c.Driver])
+// getDriverPlugin fetches the runtime driver from the plugins package
+func getDriverPlugin(c *config) (*plugin.RevaPlugin, error) {
+	plugin, err := c.load()
+	if err != nil {
+		return nil, err
 	}
+	return plugin, nil
+}
 
-	return nil, errtypes.NotFound(fmt.Sprintf("driver %s not found for user manager", c.Driver))
+func getDriver(c *config) (user.Manager, *plugin.RevaPlugin, error) {
+	isAlphaNum := regexp.MustCompile(`^[A-Za-z0-9]+$`).MatchString
```

Comment:
@ishank011, I am adding the regex check here. So when we have to load from the registry, the configuration should just be a single word (like `json`, `s3` etc). Otherwise, when we'd want to load from the plugin package the user will provide something like, `./json` or `github.com/reva-pkg/user`.

Reply:
Perfect! Will go through it

Reply:
Let's delegate this to the plugin pkg. We should do something like

```
func getDriver(c *config) (user.Manager, *plugin.RevaPlugin, error) {
	p, err := plugin.Load("userprovider", c.Driver)
	if err == nil {
	    // assert the loaded plugin into required interface
	    manager, ok := plugin.Plugin.(user.Manager)
	   ...
	} else if IsNotFound(err) {
	    // plugin not found, fetch the driver from the in-memory registry
	    if f, ok := registry.NewFuncs[c.Driver]; ok {
	    	    mgr, err := f(c.Drivers[c.Driver])
		    return mgr, nil, err
	    }
	} else {
	    return nil, nil, err
	}
	return nil, nil, errtypes.NotFound(fmt.Sprintf("driver %s not found for user manager", c.Driver))
}
```

Also, have the regex as a global var so we don't have to compile it every time